engine/src/main/java/org/kigalisim/engine/serializer/EngineResult.java

```diff
@@ -0,0 +1,231 @@
+   * @param populationValue The population value in terms of equipment
+   * @param populationNew The amount of new equipment added this year
+   * @param rechargeEmissions The greenhouse gas emissions from recharge activities
+   * @param eolEmissions The greenhouse gas emissions from end-of-life equipment
+   * @param energyConsumption The energy consumption value
+   * @param importSupplement The supplemental import data needed for attribution
+   */
+  public EngineResult(String application, String substance, int year,
+                     EngineNumber manufactureValue, EngineNumber importValue,
+                     EngineNumber recycleValue, EngineNumber domesticConsumptionValue,
+                     EngineNumber importConsumptionValue, EngineNumber recycleConsumptionValue,
+                     EngineNumber populationValue, EngineNumber populationNew,
+                     EngineNumber rechargeEmissions, EngineNumber eolEmissions,
+                     EngineNumber energyConsumption, ImportSupplement importSupplement) {
+    this.application = application;
+    this.substance = substance;
+    this.year = year;
+    this.manufactureValue = manufactureValue;
+    this.importValue = importValue;
+    this.recycleValue = recycleValue;
+    this.domesticConsumptionValue = domesticConsumptionValue;
+    this.importConsumptionValue = importConsumptionValue;
+    this.recycleConsumptionValue = recycleConsumptionValue;
+    this.populationValue = populationValue;
+    this.populationNew = populationNew;
+    this.rechargeEmissions = rechargeEmissions;
+    this.eolEmissions = eolEmissions;
+    this.energyConsumption = energyConsumption;
+    this.importSupplement = importSupplement;
+  }
+
+  /**
+   * Get the application.
+   *
+   * @return The application
+   */
+  public String getApplication() {
+    return application;
+  }
+
+  /**
+   * Get the substance.
+   *
+   * @return The substance
+   */
+  public String getSubstance() {
+    return substance;
+  }
+
+  /**
+   * Get the year the result is relevant to.
+   *
+   * @return The year
+   */
+  public int getYear() {
+    return year;
+  }
+
+  /**
+   * Get the manufacture value.
+   *
+   * @return The manufacture value in volume like kg
+   */
+  public EngineNumber getManufacture() {
+    return manufactureValue;
+  }
+
+  /**
+   * Get the import value.
+   *
+   * @return The import value in volume like kg
+   */
+  public EngineNumber getImport() {
+    return importValue;
+  }
+
+  /**
+   * Get the recycle value.
+   *
+   * @return The recycle value in volume like kg
+   */
+  public EngineNumber getRecycle() {
+    return recycleValue;
+  }
+
+  /**
+   * Get the total consumption without recycling.
+   *
+   * @return The consumption value in tCO2e or similar
+   */
+  public EngineNumber getConsumptionNoRecycle() {
+    if (!domesticConsumptionValue.getUnits().equals(importConsumptionValue.getUnits())) {
```

Comment:
We need to use `domesticConsumptionValue.getUnits()` in multiple places. Let’s put that into a variable:

```
String domesticUnits = domesticConsumptionValue.getUnits();
String importUnits = importConsumptionValue.getUnits();
```

Please add these local variables to simplify some statements.